lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -549,6 +549,9 @@ data ApiTransaction (n :: NetworkDiscriminant) = ApiTransaction
     , depth :: !(Maybe (Quantity "block" Natural))
     , direction :: !(ApiT Direction)
     , inputs :: ![ApiTxInput n]
+      -- TODO: Investigate whether the list of outputs should be non-empty, and
+      -- if so, whether the 'outputs' field can be encoded as a non-empty list.
+      -- See: https://jira.iohk.io/browse/ADP-400
```

Comment:
I don't think using NonEmpty list buys us anything here. The ApiTransaction is derived from the core type which is already valid.
Best to either address the TODO right now or remove it.

Reply:
> I don't think using NonEmpty list buys us anything here.

Using `NonEmpty` gives the reader strong evidence that this list cannot be empty. Without this, readers of the code are left wondering whether or not the list really can be empty. The current type is also at odds with the spec, which specifies that a `transactionOutputs` cannot be empty.

OTOH, if this list _can_ be empty, then I think we should at least have an comment here explaining _why_ the list can be empty, especially when _all_ the other API types involving payments use `NonEmpty`:

- `ApiSelectCoinsData`
- `ApiCoinSelection`
- `PostTransactionData`
- `PostTransactionFeeData`

I did actually have a go at changing this to use a `NonEmpty` list, but it became clear that this was a non-trivial change. Hence, I added this comment, which leaves us a reminder to tackle this in future.

Reply:
I am planning to make that thing non-empty in a very near future actually, but under special conditions that I intend to specify indeed :smile: (when creating a withdrawal for instance).